src/Interpreters/Access/InterpreterCreateUserQuery.cpp

```diff
@@ -2,31 +2,143 @@
 #include <Parsers/Access/ASTCreateUserQuery.h>
 #include <Parsers/Access/ASTRolesOrUsersSet.h>
 #include <Parsers/Access/ASTUserNameWithHost.h>
+#include <Parsers/Access/ASTAuthenticationData.h>
 #include <Parsers/ASTDatabaseOrNone.h>
 #include <Access/AccessControl.h>
 #include <Access/ContextAccess.h>
 #include <Access/User.h>
 #include <Interpreters/Access/InterpreterSetRoleQuery.h>
 #include <Interpreters/Context.h>
 #include <Interpreters/executeDDLQueryOnCluster.h>
+#include <Storages/checkAndGetLiteralArgument.h>
 #include <boost/range/algorithm/copy.hpp>
 
+#include "config.h"
+
+#if USE_SSL
+#     include <openssl/crypto.h>
+#     include <openssl/rand.h>
+#     include <openssl/err.h>
+#endif
 
 namespace DB
 {
 namespace ErrorCodes
 {
     extern const int BAD_ARGUMENTS;
+    extern const int SUPPORT_IS_DISABLED;
+    extern const int OPENSSL_ERROR;
 }
 namespace
 {
+    AuthenticationData makeAuthenticationData(const ASTAuthenticationData & query, ContextPtr context, bool check_password_rules)
```

Comment:
Initially, I added functions `makeAuthenticationData()` and `makeASTAuthenticationData()` as methods to ASTAuthenticationData, but adding Context to clickhouse_parsers breaks the standalone keeper build, so I moved them to Interpreters.

Any suggestions on where to better put them are welcome